frontend/app/src/components/settings/controls/SettingsOption.vue

```diff
@@ -57,24 +87,61 @@ export default defineComponent({
       const func = get(transform);
       const settingKey = get(setting);
       const settingValue = func ? func(newValue) : newValue;
-      const result = await updateSetting(
-        {
-          [settingKey]: settingValue
-        },
-        {
-          success: getMessage(successMessage, newValue),
-          error: getMessage(errorMessage, newValue)
+      const validators = get(rules);
+
+      let failed: boolean = false;
+
+      validators.forEach(validator => {
```

Comment:
These are vuetify-like validators, right?

Is there any specific reason why we are running these validators here and not in the component we wrap with `settings-option`. 

The original purpose of this component was update the settings and pass the success error messages to the wrapped component for display.

Reply:
Yes, it was vuetify-like validators.
But the problem I faced was, that when the vuetify components are given certain rules, it still updates the setting, even if the unallowed value is given.

With this approach, I use the same rules to also validate the value inside this component, so it won't update the setting if it breaks the rules.

Reply:
Hmm ok but this feels like an unnessessary duplication.

I am trying to think of an alternative approach here. Do you remember if `@input` fires only when a valid value is passed? At the moment we use it as part of the `v-model` and we use `@change` to hook update which fires whenever the value changes.

Reply:
@input fires every time we hit a key in the keyboard, regardless it's valid or not, so `@change` is the correct approach here.